src/components/DeleteModal.jsx

```diff
@@ -0,0 +1,106 @@
+import { Dialog, Transition } from '@headlessui/react';
+import { ExclamationIcon } from '@heroicons/react/outline';
+import React, { Fragment, useRef } from 'react';
+
+const DeleteModal = (props) => {
+  const handleDelete = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.id !== props.id) {
+        return comment;
+      } else {
```

Comment:
you won't need the else statement now, we filter before setting the new state

Reply:
removed